src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -662,6 +667,17 @@ def __call__(
                 second element is a list of `bool`s indicating whether the corresponding generated image contains
                 "not-safe-for-work" (nsfw) content.
         """
+        # We do this first to capture the "True" call values. If we do this at a later point in time,
+        # we cannot ensure that the call values weren't changed during the process.
+        # We update the `generator` later, though as we define a new generator in case it was passed as `None`.
+        if generator is None:
```

Comment:
Small problem here is also that this will break use cases where people make the run deterministic with `torch.manual` seed - is there a way to just retrieve which seed is currently used in torch? 

E.g. can we ensure that:

```py
torch.manual_seed(0)
image = pipe("a cat")
```

still gives exactly the same output?